File: Modul 4/soal_3/PRAK403-2410817310003-Muhammad_Guntur_Ricky_Adhitya.c
#include <stdio.h>

int main(){
    int num_1, num_2;
    scanf("%d %d", &num_1, &num_2);

    int state = num_1 > num_2 ? 1 : 0;
    int compare_ii = state ? (num_2-1) : (num_2+1);
    int compare_jj = state ? (num_1+1) : (num_1-1);

    for(int i = num_1, j = num_2; 1; state ? ++j && --i : --j && ++i){
        if(i == j || state ? i < j : i > j){
            printf("%d %d", i, j);
            for(int ii = state ? i-1: i+1, jj = state ? j+1: j-1; (ii != compare_ii) && (jj != compare_jj) ; state ? ++jj && --ii : --jj && ++ii){
                printf(" - %d %d", ii, jj);
            }
            break;
        }else{
            printf("%d %d - ", i, j);
        }
    }
}

File: Modul 4/soal_3/PRAK403-2410817310003-Muhammad_Guntur_Ricky_Adhitya.py
num_1, num_2 = map(int, input().split())

state = 1 if num_1 > num_2 else 0

compare_ii = num_2-1 if state else num_2+1
compare_jj = num_1+1 if state else num_1-1

i = num_1
j = num_2
while True :
    if (num_1 == num_2) or (i < j if state else i > j ) :
        print("%d %d"%(i, j), end="")
        ii = i-1 if state else i+1
        jj = j+1 if state else j-1
        
        while(ii != compare_ii) and (jj != compare_jj):
            print(" - %d %d "%(ii, jj), end="")
            ii = ii-1 if state else ii+1
            jj = jj+1 if state else jj-1
        break
    else:
        print("%d %d - "%(i, j), end="")
        i = i-1 if state else i+1
        j = j+1 if state else j-1
